test/tests/functional/pbs_hook_alarm_large_multinode_job.py

```diff
@@ -0,0 +1,224 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2017 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestPbsHookAlarmLargeMultinodeJob(TestFunctional):
+
+    """
+    This test suite contains hook test to verify that a large
+    nulti-node job does not slow down hook execution to cause an alarm.
+    """
+
+    def setUp(self):
+
+        TestFunctional.setUp(self)
+
+        a = {'resources_available.mem': '1gb',
+             'resources_available.ncpus': '1'}
+        self.server.create_vnodes(self.mom.shortname, a, 5000, self.mom)
+
+    def test_begin_hook(self):
+        """
+        Create an execjob_begin hook, import a hook content with a small
+        alarm value, and test it test it against a large multi-node job.
+        """
+        hook_name = "testhook"
+        hook_event = "execjob_begin"
+        hook_body = """
+import pbs
+e=pbs.event()
+pbs.logmsg(pbs.LOG_DEBUG, "executing begin hook %s" % (e.hook_name))
+"""
+        a = {'event': hook_event, 'enabled': 'True',
+             'alarm': '15'}
+        self.server.create_import_hook(hook_name, a, hook_body)
+
+        j = Job(TEST_USER)
+        a = {'Resource_List.select': '5000:ncpus=1:mem=1gb',
+             'Resource_List.walltime': 10}
+
+        j.set_attributes(a)
+        j.set_sleep_time(10)
```

Comment:
I've decided to follow your suggestion and not do the sleep(10) for the job but just take the default. For end and epilogue hooks tests, the walltime of 10 actually kills the job.